### PR TITLE
feat/cards

### PR DESCRIPTION
**Pull request by eginwong**
#24

### CODE REVIEW

**Comment by eginwong** [comment]
Ah, realize I am missing the special rent rules. Have to do those still.

EDIT: pushed the missing changes

File: entities/BoardService.js [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
 module.exports = class BoardSevice {
   static normalizePlayerBoardPosition(gameState) {
-    return gameState.currentPlayer.position % getProperties(gameState).length;
+    return (
+      gameState.currentPlayer.position %
+      PropertyManagementService.getProperties(gameState).length
+    );
+  }
+
+  static nearestPropertyByGroupToPlayer(gameState, propertyGroup) {
```
Comment (DivvyByZero) [inline entities/BoardService.js:12]: Suggestion: Specify "Forward-Only Search"

File: test/BoardService.test.js [hunk 2]
```diff
@@ -30,4 +30,73 @@ describe('BoardService', () => {
       );
     });
   });
+
+  describe('nearestPropertyByGroupToPlayer', () => {
+    it('should return property closest to player by property group when property is before GO', () => {
+      gameState.currentPlayer.position = 10;
```
Comment (DivvyByZero) [inline test/BoardService.test.js]: What is this ?

File: test/BoardService.test.js [hunk 3]
```diff
@@ -30,4 +30,73 @@ describe('BoardService', () => {
...
+        gameState,
+        propertyGroup
+      );
+      expect(actualPropertyReturned).to.equal(
+        pennsylvaniaRailroadProperty,
+        `Incorrectly returns nearest property by property group ${actualPropertyReturned}`
+      );
+    });
+    it('should return property closest to player by property group when property is after GO', () => {
+      gameState.currentPlayer.position = 39;
```
Comment (DivvyByZero) [inline test/BoardService.test.js]: What is this ?

File: test/Components/Deck.test.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        cards.length,
+        'draw did not remove a card from the input'
+      );
+    });
+  });
+  describe('discard', () => {
+    it('should add the input card to the second input of discarded cards', () => {
+      const discardedCards = [1, 2, 3].map((x) => ({ id: x }));
+
+      const output = Deck.discard({ id: 4 }, discardedCards);
```
Comment (DivvyByZero) [inline test/Components/Deck.test.js]: Can we put this into a constant ? `const TEST_CARD = { id: 4 }`

File: test/Rules/INCOME_TAX.test.js [hunk 5]
```diff
@@ -134,6 +134,22 @@ describe('Rules -> INCOME_TAX', () => {
         }% of networth when VARIABLE was chosen for payment `
       );
     });
+    it('should decrease player cash by config incomeTaxRate if player chooses VARIABLE, rounded to 2 decimal places', () => {
+      const startingCash = gameState.currentPlayer.cash;
+      const assets = 11;
```
Comment (DivvyByZero) [inline test/Rules/INCOME_TAX.test.js:139]: `// arbitary value`

**Review by DivvyByZero — APPROVED** [review]
Approvèd

**Pull request merged**